pages/admin/dashboard/tools.tsx

```diff
@@ -1,54 +1,95 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
 import {
     Stack,
     Flex,
     Text,
     Button,
     useDisclosure,
     Spacer,
+    Spinner,
 } from "@chakra-ui/react";
 import { ToolCard, Modal, AdminLayout } from "@components";
 import { Page } from "types/Page";
 import axios from "axios";
+import useSWR from "swr";
+
+const fetcher = async (url) => {
+    const response = await axios({
+        method: "GET",
+        url,
+    });
+    return response.data;
+};
```

Comment:
this fetcher function is the same for the modules page - should we put this in another directory (like a utils folder) so it can be reused